### DISCUSSION TITLE
Newbie. Don't understand how to use APP LAYOUT fully.

### DISCUSSION THREAD

**Discussion by Smithology** [rxhanson/RectanglePro-Community#271] [Q&A]
Hi,
I bought RP specifically for the App Layout feature as I understood it. That is I can position i.e. 2 apps on DESKTOP 1 and then apply a KB Command that will always bring me to DESKTOP 1 with those 2 apps arranged where I left them.
I'm using Latest Mac OS Ventura (M2 Mini) with 34" widescreen. I use 3-4 SPACES all the time. I've used the menu SAVE CURRENT LAYOUT and given it the shortcut ⌘ ⌥ 2

I've been looking through the info on github, but I'm not a coder and this is my first experience with any type of very customizable program for MAC OS. I've even had some difficulty understanding what I've read about other topics, i.e., STASHING. Again, I got it to work most basically, but then returning the stashed windows to where they were didn't happen for me. They came back either full-screen or centered. Anyway, for another thread.

What am I misunderstanding? Thoughts?
Thank you.
Michael

MORE INFO from OP:
I did have Stage Manager engaged when starting, but have turned it off and continued to experiment/use  RP the same way 

![Screenshot 2023-05-07 at 2 25 15 PM](https://user-images.githubusercontent.com/132843030/236695875-af7b0afa-5358-4396-b1a5-4ee90dce9c7e.jpg)

**Comment by rxhanson** [comment]
Thanks for purchasing. 

The App Layout is only for saving/applying a layout of windows to your current space, not for switching spaces.

Apple doesn't really give 3rd party developers much in the way of working with spaces, hence Rectangle Pro not doing much with them.

I have considered adding a way to switch spaces, like a configurable keyboard shortcut, although that may or may not be built into the App Layouts functionality. If you have a space laid out the way you want with an app layout, then sounds like what you're looking for at that point is a way to switch back to that space quickly? There's also this: https://apple.stackexchange.com/questions/213549/keyboard-shortcuts-for-switching-spaces

On stashing, if you want to make a screen recording so I can understand what you're experiencing vs your expectations, that might be the most helpful for me. 

I am starting to make demo/tutorial videos, although I only have one out at the moment, about the window throw: https://www.youtube.com/@rectangleapp

**Reply by panckreous** [reply]
im very much looking forward to the youtube videos

**Comment by Smithology** [comment]
Wow,
Thanks for the quick personal response. We’re almost talking about the same thing.
I’m not looking for RP to switch places as a separate function, I just thought if I were on Space 2, using my assigned App Layout command would take me to 1. That’s not a problem. 

However, I thought once I SAVED the layout, I can apply the KB Shortcut, while on the space I saved the layout, the windows would snap back to their saved positions. That hasn’t worked for me. I moved 1 app a few inches and it doesn’t return to the location when saved. However, I’m not sure if this happened. I moved 1 app to test. Went to work on another Space. Came back to Space 1 and the moved app did return. Again, not sure….

I’ll work in getting you a screen video of stashing. Thank you again very much.
MB
Looking forward to more YT videos.

**Comment by k4u5hik** [comment]
I tried the app today for the first time and straightaway bought it. It's great! Thank you!

EDIT: Just realised that I can use of at most three copies of the product, on at most three computers at any given time. Will be recommending it. Appreciate it!

**Reply by rxhanson** [reply]
Thanks! Glad you like it :)

**Comment by Smithology** [comment]
Hello,
I am a registered user.

I have 4 APPLAYOLUTS saved that are identical except they are assigned to different SPACES.
1-3 work exactly as expected, however the 4th is not. When I press the KB Shortcut nothing occurs.
Is there a limit someway. Thoughts. Pic Attached.

Using PIC1 you can see there are 2 instances of Messages. I don’t know how/why the 2nd is there. I’ve attached a pic that shows another instances or window isn’t running.
Thoughts?
Thanks,
MB

**Reply by rxhanson** [reply]
Thanks for purchasing! Looks like your image didn’t come through. It might be easiest if you export your config and I can import it and look at it.

**Comment by Smithology** [comment]
Here's config
[RectangleProConfig.json](https://github.com/rxhanson/RectanglePro-Community/files/13812022/RectangleProConfig.json)

**Reply by Smithology** [reply]
[RectangleProConfig.json](https://github.com/rxhanson/RectanglePro-Community/files/13875814/RectangleProConfig.json)

**Reply by rxhanson** [reply]
Thanks, sorry for the delay. From my testing, it looks like this is dependent on the windows for the VPN apps not responding properly to the macOS accessibility API requests. Unfortunately, this isn't something that I have a workaround for at this time. I'll post back here if I figure out a workaround.

**Reply by Smithology** [reply]
Thanks, that makes sense. Also, good for you know. Thanks.
MB

**Comment by ssnkhan** [comment]
Just completing my trial, and had a question about saved app layouts. Is there anyway of triggering these without running a shortcut? Double clicking on the name has no effect.

**Reply by rxhanson** [reply]
You can also use the Window Throw / Long Throw to trigger them, and you can add them to the menu bar menu from the Menu & Icon tab in the Settings window.
